data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -1309,6 +1309,9 @@ private void buildUpdateStatement(
                 queryString.append(COMMA);
             }
         }
+        if(queryString.charAt(queryString.length() - 1) == COMMA){
+            queryString.deleteCharAt(queryString.length() - 1);
+        }
```

Comment:
This is happening when the last field  `prop.isGenerated()`.  It is better to add at last instead of polluting everywhere by checking `prop.isGenerated()`.

Reply:
Perhaps move the appending of the comma to the start of the loop and check for i == 0? Only append to i after the if check for null and generated. The other `queryString.append(COMMA)` could likely be removed then as well

Reply:
This would be better yes

Reply:
My previous implementation was buggy as well for extreme cases. I think this one is better. I appended comma before 
 `queryString.append(columnName).append('=');` only and guarded it with boolean. 

Simple adding a `i == 0` will be not working because of inner loop. There is 3 places adding comma in the method.